rados/src/main/java/com/yahoo/ycsb/db/RadosClient.java

```diff
@@ -0,0 +1,167 @@
+/**
+ * Copyright (c) 2016 YCSB contributors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+
+package com.yahoo.ycsb.db;
+
+import com.ceph.rados.Rados;
+import com.ceph.rados.IoCTX;
+import com.ceph.rados.jna.RadosObjectInfo;
+import com.ceph.rados.ReadOp;
+import com.ceph.rados.ReadOp.ReadResult;
+import com.ceph.rados.exceptions.RadosException;
+
+
+import com.yahoo.ycsb.ByteIterator;
+import com.yahoo.ycsb.DB;
+import com.yahoo.ycsb.DBException;
+import com.yahoo.ycsb.Status;
+import com.yahoo.ycsb.StringByteIterator;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map.Entry;
+import java.util.Properties;
+import java.util.Set;
+import java.util.Vector;
+
+import org.json.JSONObject;
+
+/**
+ * YCSB binding for <a href="http://ceph.org/">RADOS of Ceph</a>.
+ *
+ * See {@code rados/README.md} for details.
+ */
+public class RadosClient extends DB {
+
+  private static Rados rados;
+  private static IoCTX ioctx;
+
+  public static final String CONFIG_FILE_PROPERTY = "rados.configfile";
+  public static final String CONFIG_FILE_DEFAULT = "/etc/ceph/ceph.conf";
+  public static final String ID_PROPERTY = "rados.id";
+  public static final String ID_DEFAULT = "admin";
+  public static final String POOL_PROPERTY = "rados.pool";
+  public static final String POOL_DEFAULT = "data";
+
+  public void init() throws DBException {
+    Properties props = getProperties();
+
+    String configfile = props.getProperty(CONFIG_FILE_PROPERTY);
+    if (configfile == null) {
+      configfile = CONFIG_FILE_DEFAULT;
+    }
+
+    String id = props.getProperty(ID_PROPERTY);
+    if (id == null) {
+      id = ID_DEFAULT;
+    }
+
+    String pool = props.getProperty(POOL_PROPERTY);
+    if (pool == null) {
+      pool = POOL_DEFAULT;
+    }
+
+    rados = new Rados(id);
```

Comment:
No, I mean the member variables `rados` and `ioctx` are static. We instantiate a RadosClient per worker thread in the client, so access to initializing and cleaning up these static members must be protected by a lock.

If the member classes themselves aren't threadsafe, then you'll need to either guard all uses with a lock or switch to making them non-static.